### PR TITLE
fix: lint does not work on new Otter webapp

### LINKED ISSUE #1661: [Bug]: Linter does not work on a new Otter webapp in a monorepo
### Package name

eslint-config-otter

### Package version

10.1.5

### Reproduction steps

```shell
npm create @o3r otter-project
cd otter-project
ng g application webapp
# choose preset BASIC, no Jest and no Playwright when prompted
ng add @o3r/eslint-config-otter
npm run ng lint
```

### Current result

```shell
$ npm run ng lint webapp

> otter-project@0.0.0 ng
> ng lint webapp

Linting "webapp"...
An unhandled exception occurred: Invalid lint configuration. Nothing to lint. Please check your lint target pattern(s).
See "/private/var/folders/y9/zqm_33k537sd7lbz_b35cn780000gp/T/ng-Rifhs8/angular-errors.log" for further details.
```

### Expected result

Linting works.

### Additional comments

`ng add @o3r/eslint-config-otter` create some eslint specific files at the root of the project and not in the webapp

### PR DESCRIPTION
**Pull request by vscaiceanu-1a**
## Related issues

- :bug: Fixes #1661

### CODE REVIEW

File: packages/@o3r/eslint-config-otter/schematics/ng-add/linter/templates/tsconfig.eslint.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
-  "extends": "./tsconfig.app",
+  "extends": "./tsconfig.json",
   "include": [
     ".eslintrc.js",
     "jest.config.js",
-    "src/**/*.ts",
-    "e2e-playwright/**/*.ts"
+    "**/src/**/*.ts",
+    "**/e2e-playwright/**/*.ts"
```
Comment (kpanot) [inline packages/@o3r/eslint-config-otter/schematics/ng-add/linter/templates/tsconfig.eslint.json:7]: Not sure it is required as the purpose of this file is to add file not included in build process

**Review by cpaulve-1A — DISMISSED** [review]

File: packages/@o3r/eslint-config-otter/schematics/ng-add/linter/templates/tsconfig.eslint.json [hunk 2]
```diff
@@ -1,9 +1,8 @@
 {
-  "extends": "./tsconfig.app",
+  "extends": "./tsconfig.json",
   "include": [
     ".eslintrc.js",
     "jest.config.js",
-    "src/**/*.ts",
-    "e2e-playwright/**/*.ts"
+    "**/src/**/*.ts"
```
Comment (kpanot) [inline packages/@o3r/eslint-config-otter/schematics/ng-add/linter/templates/tsconfig.eslint.json:6]: Not needed as well

Comment (vscaiceanu-1a) [reply packages/@o3r/eslint-config-otter/schematics/ng-add/linter/templates/tsconfig.eslint.json:6]: It is actually needed
```shell
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/apps/webapp/src/main.ts` using `parserOptions.project`: test-eslint-basic-2/tsconfig.eslint.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file
```

Comment (kpanot) [reply packages/@o3r/eslint-config-otter/schematics/ng-add/linter/templates/tsconfig.eslint.json:6]: This error is different, it just mean that you should register both `tsconfig.json` and `tsconfig.eslint.json` in the ts-eslint projects field

Comment (vscaiceanu-1a) [reply packages/@o3r/eslint-config-otter/schematics/ng-add/linter/templates/tsconfig.eslint.json:6]: `tsconfig.json` doesn't have any path mapping. Let's discuss this point tomorrow morning.

File: packages/@o3r/eslint-config-otter/schematics/index.it.spec.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+describe('new otter application with eslint-config', () => {
+  test('should add eslint-config to existing application', () => {
+    const { workspacePath, projectName, isInWorkspace, o3rVersion } = o3rEnvironment.testEnvironment;
+    const execAppOptions = {...getDefaultExecSyncOptions(), cwd: workspacePath};
+    packageManagerExec({script: 'ng', args: ['add', `@o3r/eslint-config-otter@${o3rVersion}`, '--skip-confirmation', '--project-name', projectName]}, execAppOptions);
+
+    expect(() => packageManagerInstall(execAppOptions)).not.toThrow();
+    expect(() => packageManagerRunOnProject(projectName, isInWorkspace, {script: 'build'}, execAppOptions)).not.toThrow();
+    expect(() => packageManagerExec({script: 'ng', args: ['lint', projectName]}, execAppOptions)).toThrow('Lint errors found in the listed files');
```
[the remark this answers is not in the document]
Comment (vscaiceanu-1a) [reply packages/@o3r/eslint-config-otter/schematics/index.it.spec.ts:23]: At this point, yes. The goal is to be able to run the linter.
In a future PR I plan to 
1. fix all linter errors that are not fixable with `lint --fix`
2. run the `applyLintFix` after app and lib generation and
3. write IT tests to make sure there are no lint errors for apps and libs

[the remark this answers is not in the document]
Comment (vscaiceanu-1a) [reply packages/@o3r/eslint-config-otter/schematics/index.it.spec.ts:23]: https://github.com/AmadeusITGroup/otter/issues/1690

File: packages/@o3r/eslint-plugin/tsconfig.spec.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
     }
   ],
   "include": [
-    "./src/**/*.spec.ts",
+    "./src/**/*.spec.ts"
```
Comment (kpanot) [inline packages/@o3r/eslint-plugin/tsconfig.spec.json:9]:
```suggestion
    "**/*.spec.ts"
```

**Review by fpaul-1A — DISMISSED** [review]

File: packages/@o3r/eslint-config-otter/schematics/index.it.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (vscaiceanu-1a) [reply packages/@o3r/eslint-config-otter/schematics/index.it.spec.ts]: The purpose was to make sure the linter is running. I have adapted the condition to test that the previous error is no longer thrown.

File: packages/@o3r/eslint-config-otter/schematics/ng-add/linter/index.ts [hunk 5]
```diff
@@ -16,34 +16,77 @@ export function updateLinterConfigs(options: { projectName?: string | null | und
...
 
-    if (eslintExists) {
+    if (tree.exists(eslintFilePath)) {
       const eslintFile = tree.readJson(eslintFilePath) as { extends?: string | string[] };
       eslintFile.extends = eslintFile.extends ? (eslintFile.extends instanceof Array ? eslintFile.extends : [eslintFile.extends]) : [];
 
-      if (eslintFile.extends.indexOf(tsEslintParserDep) === -1) {
-        eslintFile.extends.push(tsEslintParserDep);
+      const eslintConfigOtter = '@o3r/eslint-config-otter';
+      if (eslintFile.extends.indexOf(eslintConfigOtter) === -1) {
```
[the remark this answers is not in the document]
Comment (vscaiceanu-1a) [reply packages/@o3r/eslint-config-otter/schematics/ng-add/linter/index.ts:25]: it is a corner case that isn't currently covered

**Review by kpanot — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

File: packages/@o3r/eslint-config-otter/schematics/ng-add/linter/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (vscaiceanu-1a) [reply packages/@o3r/eslint-config-otter/schematics/ng-add/linter/index.ts:25]: #1695

**Pull request merged**